scalding-core/src/main/scala/com/twitter/scalding/Job.scala

```diff
@@ -214,7 +210,14 @@ class Job(val args: Args) extends FieldConversions with java.io.Serializable {
 
   private def executionContext: Try[ExecutionContext] =
     Config.tryFrom(config).map { conf =>
-      ExecutionContext.newContext(conf)(flowDef, mode)
+      val updatedConfig = if (conf.getRequireOrderedSerialization) {
+        state.asScala.foldLeft(conf) {
```

Comment:
ExecutionApp does not hit this code path. We should go deeper and look inside ExecutionContext.run to get to the shared code path.